endpoint.go

```diff
@@ -615,16 +637,29 @@ func (ep *endpoint) Delete() error {
 	ep.Lock()
 	epid := ep.id
 	name := ep.name
-	sb, _ := n.getController().SandboxByID(ep.sandboxID)
-	if sb != nil {
-		ep.Unlock()
+	sbid := ep.sandboxID
+	locator := ep.locator
+	ep.Unlock()
+
+	if force {
+		if err = n.validateForceDelete(locator); err != nil {
+			return fmt.Errorf("unable to force delete endpoint %s : %v", name, err)
+		}
+	}
```

Comment:
Can we add a `defer ep.ReleaseAddress()` here if validation succeeds so that we release the IP address no matter what happens in this function? Which means the `ep.ReleaseAddress()` call in the end of the function should only be called in non-force case.